common/src/java/com/zimbra/common/soap/W3cDomUtil.java

```diff
@@ -281,18 +285,41 @@ public static Document parseXMLToDoc(String xml)
         }
     }
 
+    public static ByteArrayInputStream removeInvalidXMLChars(byte[] bytes) {
+        String xml = new String(bytes);
+        Pattern xmlInvalidChars = Pattern.compile(LC.xml_invalid_chars_regex.value());
+        xml = xmlInvalidChars.matcher(xml).replaceAll("");
+        ByteArrayInputStream is = new ByteArrayInputStream(xml.getBytes());
+        return is;
+    }
+
     public static Document parseXMLToDoc(InputStream is)
     throws XmlParseException {
+        byte[] bytes = null;
+        ByteArrayInputStream bais = null;
         javax.xml.parsers.DocumentBuilder jaxbBuilder = getBuilder();
         jaxbBuilder.reset();
         jaxbBuilder.setErrorHandler(new JAXPErrorHandler());
         try {
-            return jaxbBuilder.parse(is);
+            bytes = ByteStreams.toByteArray(is);
+            bais = new ByteArrayInputStream(bytes);
+            return jaxbBuilder.parse(bais);
         } catch (SAXException | IOException e) {
-            /* Bug 93816 log actual problem but throw generic one to avoid information disclosure */
-            logParseProblem(e);
-            throw XmlParseException.PARSE_ERROR();
+            if (e instanceof SAXException) {
+                try {
+                    ZimbraLog.backup.debug("SAXException for XML parsing, removing invalid characters and parsing again.");
```

Comment:
Also log the exception

Reply:
Exception will get logged if its still fails after removing the invalid xml chars.

Reply:
fixed